Pokedex_ios/ViewModel/PokemonListViewModel.swift

```diff
@@ -28,12 +29,25 @@ class PokemonListViewModel: NSObject {
         pokedexService.getPokemons { result in
             switch result {
             case .success(let pokemons):
-                self.pokemons = pokemons
+                self.pokemonsOriginalList = pokemons
+                self.pokemons = self.pokemonsOriginalList
             case .failure( _):
                 self.showErrorAlert?()
             }
         }
     }
+    
+    func searchPokemonsByName(searchText: String) {
```

Comment:
searchPokemonByName should be called inside a global method for applying filters.
Like this:
applyFilters(){
   this.pokemons = pokemonOriginalList
   // Now here:   
  this.searchPokemonByName() //empty parameters because search text should be a viewmodel variable.
    
   
}
}